nntrainer/optimizers/lr_scheduler_linear.cpp

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2024 Hyunwoo LEE <dlgusdn0414@snu.ac.kr>
+ *
+ * @file   lr_scheduler_linear.cpp
+ * @date   11 November 2024
+ * @brief  This is Linear Learning Rate Scheduler class
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Hyunwoo LEE <dlgusdn0414@snu.ac.kr>
+ * @bug    No known bugs except for NYI items
+ *
+ */
+
+#include <cmath>
+
+#include <common_properties.h>
+#include <lr_scheduler_linear.h>
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <node_exporter.h>
+
+namespace nntrainer {
+
+LinearLearningRateScheduler::LinearLearningRateScheduler() :
+  lr_props(props::MaxLearningRate(), props::MinLearningRate(),
+           props::DecaySteps()) {}
+
+void LinearLearningRateScheduler::finalize() {
+  NNTR_THROW_IF(std::get<props::MaxLearningRate>(lr_props).empty(),
+                std::invalid_argument)
+    << "[LinearLearningRateScheduler] Max Learning Rate is not set";
+  NNTR_THROW_IF(std::get<props::MinLearningRate>(lr_props).empty(),
+                std::invalid_argument)
+    << "[LinearLearningRateScheduler] Min Learning Rate is not set";
+  NNTR_THROW_IF(std::get<props::DecaySteps>(lr_props).empty(),
+                std::invalid_argument)
+    << "[LinearLearningRateScheduler] Decay Steps is not set";
```

Comment:
it would be nice if you also check that decay_steps is positive.

Reply:
Added it